src/Data/Nat.agda

```diff
@@ -42,5 +42,14 @@ open import Data.Nat.Properties public
 
 -- Version 0.17
 
+-- Version 2.0
+
 open import Data.Nat.Properties public
-  using (≤-pred)
+  hiding (≤-pred; <-pred)
+
+≤-pred : ∀ {m n} → (m ≤ n) → pred m ≤ pred n
+≤-pred {m = zero}              _   = z≤n
+≤-pred {m = suc _} {n = suc _} m≤n = s≤s⁻¹ m≤n
+
+<-pred : ∀ {m n} → .⦃ _ : NonZero m ⦄ → (m < n) → pred m < pred n
+<-pred {m = suc _} {n = suc _} = s<s⁻¹
```

Comment:
Re: our discussion this morning.

* previously, these lemmas (but with different types) were being exported here from `Data.Nat.Properties`...
* now, those lemmas are (still) defined in `Data.Nat.Properties`, but now deprecated there, so if they are used on import from that module, then they will trigger a deprecation warning, as they would have done by being imported and used from here
* the change now proposed is to continue to deprecate them in `Data.Nat.Properties`, but to allow a non-deprecated export of lemmas with those names from `Data.Nat`, but now with different types, which reflect the 'monotone/congruent' formulation of the lemmas.

So, I've probably made a mistake giving them the same name as the former lemmas which are now deprecated. Perhaps it would be better to have *new* names for these (and then find a home for them, either in this module, or indeed in `Data.Nat.Properties`), while continuing to export the old names, but now deprecated, from `Data.Nat.Properties`.

Apologies for causing so much confusion with this change!

So it seems that there are two options:
* don't bother to have these lemmas at all
* rename them, and then they will be distinct from the deprecated names, and then find a home for them somewhere

What do others think?

Reply:
My own view is to retain the new lemmas, but rename them as
* `pred-≤-monotone` (or `...-mono`)
* `pred-<-monotone` (or `...-mono`)

and push them into `Data.Nat.Properties`, but then re-export them here.

Reply:
Ha, there already exists `pred-mono` for the first one! Sigh... for consistency, I'd want to deprecate *that* name as well. Oh dear.